hdate/htables.py

```diff
@@ -782,117 +607,181 @@ class HolidayTypes(Enum):
+    "tu_bav": LANG("Tou be'Av", "Tu B'Av", DESC('ט"ו באב', 'ט"ו באב')),
+    "yom_hashoah": LANG("Yom HaShoah", "Yom HaShoah", DESC("יום השואה", "יום השואה")),
+    "yom_hazikaron": LANG(
+        "Yom haZicaron", "Yom HaZikaron", DESC("יום הזכרון", "יום הזכרון")
+    ),
+    "yom_yerushalayim": LANG(
+        "Yom Yeroushalaïm", "Yom Yerushalayim", DESC("יום ירושלים", "יום י-ם")
+    ),
+    "shmini_atzeret": LANG(
+        "Shemini Atseret", "Shmini Atzeret", DESC("שמיני עצרת", "שמיני עצרת")
+    ),
+    "pesach_viii": LANG(
+        "Pessah VIII", "Pesach VIII", DESC("אחרון של פסח", "אחרון של פסח")
+    ),
+    "shavuot_ii": LANG("Shavouot II", "Shavuot II", DESC("שני של שבועות", "ב' שבועות")),
+    "sukkot_ii": LANG("Souccot II", "Sukkot II", DESC("שני של סוכות", "ב' סוכות")),
+    "pesach_ii": LANG("Pessah II", "Pesach II", DESC("שני של פסח", "ב' פסח")),
+    "family_day": LANG(
+        "Fête de la Famille", "Family Day", DESC("יום המשפחה", "יום המשפחה")
+    ),
+    "memorial_day_unknown": LANG(
+        "Jour du souvenir",
+        "Memorial day for fallen whose place of burial is unknown",
+        DESC("יום זכרון...", "יום זכרון..."),
+    ),
+    "rabin_memorial_day": LANG(
+        "Jour commémoratif Yitzhak Rabin",
+        "Yitzhak Rabin memorial day",
+        DESC("יום הזכרון ליצחק רבין", "יום הזכרון ליצחק רבין"),
+    ),
+    "zeev_zhabotinsky_day": LANG(
+        "Jour de Zeev Zhabotinsky",
+        "Zeev Zhabotinsky day",
+        DESC("יום ז'בוטינסקי", "יום ז'בוטינסקי"),
+    ),
+    "rosh_chodesh": LANG("Rosh Hodesh", "Rosh Chodesh", DESC("ראש חודש", "ראש חודש")),
+}
+
 
 def get_all_holidays(language: str) -> list[str]:
-    """Helper method to get all the holiday descriptions."""
-    return [
-        h.description.hebrew.long if language == "hebrew" else h.description.english
-        for h in HOLIDAYS
-    ] + ["Chanukah, Rosh Chodesh"]
+    """Helper method to get all the holiday descriptions in the specified language."""
+    holidays_list = []
+
+    for h in HOLIDAYS:
+        # Get the description entry from the dictionary using the holiday name
+        description_entry = HOLIDAY_DESCRIPTIONS.get(h.name)
+
+        if description_entry:
+            # Try to access the description in the specified language
+            description_language = getattr(description_entry, language, None)
+
+            if description_language:
+                # Check if it's an instance of DESC with 'long' and 'short'
+                if isinstance(description_language, DESC):
+                    holidays_list.append(description_language.long)
+                else:
+                    holidays_list.append(description_language)
+            else:
+                # If the specified language is not available, default to English
+                holidays_list.append(description_entry.english)
+        else:
+            # If no description is found, use the holiday name
+            holidays_list.append(h.name)
+
```

Comment:
I suggest using a short method to define what should be returned, and then use a list comprehension to create the list:

```python
def holiday_name(entry, language):
    entry_language = getattr(entry, language, None)
    if entry_language:
        return entry_language.long if isinstance(entry_language, DESC) else entry_language
    return entry.english

holidays = [holiday_name(h, language) for h in HOLIDAYS]
holidays += ["Chanukah, Rosh Chodesh"]  # Need to fix here also support for other languages
```